src/App.tsx

```diff
@@ -9,8 +11,10 @@ import type { ActivePage } from '@/types/state.ts';
 import { getCartTotalAmount } from '@/utils/getCartTotalAmount.ts';
 
 function App() {
+    const isPreference = window.matchMedia('(prefers-color-scheme: dark)').matches;
     const [activePage, setActivePage] = useState<ActivePage>('about');
     const [totalCart, setTotalCart] = useState<number>(0);
+    const [isTheme, setIsTheme] = useLocalStorage<string | boolean>('light', isPreference);
```

Comment:
add `useState` for setting theme


`const [theme, setTheme] = useState<''light|'dark'>(() => getUserTheme());`